src/main/groovy/lang/ObjectRange.java

```diff
@@ -408,107 +401,86 @@ public void step(int step, Closure closure) {
      */
     @Override
     public Iterator iterator() {
-        // non thread-safe iterator
-        final Iterator innerIterator = new StepIterator(this, 1);
-        final Iterator safeIterator = new Iterator() {
+        return new Iterator() {
+            private final Iterator delegate = new LazySteppingIterator(ObjectRange.this, 1);
+
             @Override
             public synchronized boolean hasNext() {
-                return innerIterator.hasNext();
+                return delegate.hasNext();
             }
 
             @Override
             public synchronized Object next() {
-                return innerIterator.next();
+                return delegate.next();
             }
 
             @Override
-            public void remove() {
-                innerIterator.remove();
+            public synchronized void remove() {
```

Comment:
no, don't synchronize it. Call innerIterator.remove(); like before. I know innerIterator.remove() also always throws, but still it's better to have just one location that throws.